lib/temporal/connection/serializer/start_child_workflow.rb

```diff
@@ -34,6 +35,12 @@ def serialize_headers(headers)
 
           Temporal::Api::Common::V1::Header.new(fields: object.headers)
```

Comment:
nit: This is probably out of scope for this PR, but this also needs to have `to_payload_map` call